### DISCUSSION TITLE
How to implement code in rust to update the position x and y of the component with type length

### DISCUSSION THREAD

**Discussion by CarbonPool** [slint-ui/slint#3501] [Q&A]

**Comment by tronical** [comment]
Perhaps what you're looking for is to expose the `x` and `y` properties of an element to Rust. Properties of type `length` are mapped to `f32` in Rust. For more information about the mappings, see the [Type Mappings](https://slint.dev/releases/1.2.0/docs/rust/slint/docs/type_mappings/) documentation. For information about how to expose properties and access them, check out the [Generated components](https://slint.dev/releases/1.2.0/docs/rust/slint/#generated-components) documentation.

**Reply by CarbonPool** [reply]
How to deal with pixel units such as `px`, `cm`...

**Reply by tronical** [reply]
If you assign for example `2cm` to a `length` property, we assume 96 pixels per inch, so "37.8 pixels" per cm and thus the `f32` property would have a value of `75.5` in Rust/C++.

**Reply by CarbonPool** [reply]
An idea, it would be great if slint supports it.
```
in property <int> pos-x: 0;
x: root.pos-x + "px";
```

**Reply by tronical** [reply]
Yes, this is doable by multiplication:

```
in property <int> pos-x: 0;
x: root.pos-x * 1px;
```
